File: app/src/main/java/com/mojtaba_shafaei/android/MainActivity.java
package com.mojtaba_shafaei.android;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity{

ErrorMessage em;

@Override
protected void onCreate(Bundle savedInstanceState){
  super.onCreate(savedInstanceState);
  setContentView(R.layout.activity_main);
  em = findViewById(R.id.errorMessage);
  em.hide();

  findViewById(R.id.btn_menu_more).setOnClickListener(view -> InfoActivity.startActivity(this));

  findViewById(R.id.btnShowError).setOnClickListener(view -> em.showNoData());
  findViewById(R.id.btnShowLoading).setOnClickListener(view -> em.showLoading());
  findViewById(R.id.btnShowListLoading).setOnClickListener(view -> {
    em.showListLoading();
  });

  findViewById(R.id.root).setOnClickListener(view -> em.hide());
  findViewById(R.id.root).setOnLongClickListener(view -> {
    em.setLoadingData(new AnimationData(R.raw.circle_red_button, 2.50f, .6f, 0x20000000));
    em.showLoading();
    return false;
  });
}
}

File: app/src/main/java/com/mojtaba_shafaei/android/MainActivity.kt
package com.mojtaba_shafaei.android

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.mojtaba_shafaei.android.ErrorMessage.State
import com.mojtaba_shafaei.android.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
 private lateinit var binding: ActivityMainBinding
 
 override fun onCreate(savedInstanceState: Bundle?) {
  super.onCreate(savedInstanceState)
  binding = ActivityMainBinding.inflate(layoutInflater)
  val view = binding.root
  setContentView(view)

  binding.btnSimpleShowInternetError.setOnClickListener { binding.em.state = State.internetError() { Log.e("TAG", "internet error fired") } }
  binding.btnShowInternetError.setOnClickListener { binding.em.state = State.internetError(message =binding.btnShowInternetError.tag as String? , actionTitle = "My Action...") { Log.e("TAG", "internet error 2 fired") } }

  binding.btnClose.setOnClickListener { binding.em.state = State.hidden() }

  binding.btnShowError.setOnClickListener { binding.em.state = State.error(message = getString(R.string.error_String)).copy(messageTextColor = 0xFFFF0000.toInt()) }
  binding.btnShowError2.setOnClickListener { binding.em.state = State.error(message = "Hey Developer. This is a Green error!", actionTitle = "Action Title", action = {
   Log.e("TAG", "btnShowError2 fired")
  })
          .copy(messageTextColor = 0xFF00FF00.toInt()) }// the copy method is in the parent class of State

  binding.btnShowNoDataError.setOnClickListener { binding.em.state = State.noData() }
  binding.btnShowNoDataError2.setOnClickListener { binding.em.state = State.noData(message = "Not Found...", actionTitle = "Try Again♣", action = { Log.d("TAG", "No Data2 fired")})
          .copy(mainIcon = android.R.drawable.stat_notify_error)}

  binding.btnShowLoading.setOnClickListener { binding.em.state = State.loading() }

  binding.btnShowMessage.setOnClickListener { binding.em.state = State.message(message = "Hey Developer, This is a message that you can change it.",
          action = { Log.d("TAG", "action Message fired") }) }
 }
}
